src/pyodide/internal/builtin_wrappers.js

```diff
@@ -18,6 +18,23 @@ export function monotonicDateNow() {
   return now + lastDelta;
 }
 
+/**
+ * We initialize Python at top level, but it tries to initialize the random seed with
+ * crypto.getRandomValues which will fail at top level. So we don't produce any entropy the first
+ * time around and we reseed the rng in the first request context before executing user code.
+ */
+export function getRandomValues(arr) {
+  try {
+    return crypto.getRandomValues(arr);
+  } catch (e) {
+    if (e.message.includes("Disallowed operation called within global scope")) {
+      // random.seed() can't work at startup. We'll seed again under the request scope.
+      return arr;
+    }
```

Comment:
This is fragile and will break as soon as someone changes the error message (which presumably would cause Python Workers to break).

Can we do better here without too much work? If not are we sure that our tests will fail reliably when this code breaks (would be good to test this case)?

Reply:
Python will never start without this.